src/mctpd.c

```diff
@@ -2893,6 +2893,49 @@ static int bus_link_get_prop(sd_bus *bus,
 	return rc;
 }
 
+static int notify_discovery(ctx *ctx, int ifindex)
+{
+	int rc = 0;
+	dest_phys desti = { 0 }, *dest = &desti;
+	struct mctp_ctrl_cmd_discovery_notify req = { 0 };
+	struct mctp_ctrl_resp_discovery_notify *resp;
+	uint8_t *buf;
+	size_t buf_size;
+	struct sockaddr_mctp_ext resp_addr;
+
+	dest->ifindex = ifindex;
+	assert(mctp_nl_ifaddr_byindex(ctx->nl, dest->ifindex,
+				      &dest->hwaddr_len));
+	memset(dest->hwaddr, 0, sizeof dest->hwaddr);
+
+	req.ctrl_hdr.command_code = MCTP_CTRL_CMD_DISCOVERY_NOTIFY;
+	req.ctrl_hdr.rq_dgram_inst = RQDI_REQ;
+
+	rc = endpoint_query_phys(ctx, dest, MCTP_CTRL_HDR_MSG_TYPE, &req,
+				 sizeof(req), &buf, &buf_size, &resp_addr);
```

Comment:
`endpoint_query_phys` will wait for a response, but Discovery Notify does not send a response. We'll probably need a different helper for datagram-style messages.